code/controller/actuator.cpp

```diff
@@ -22,152 +23,147 @@ Actuator::Actuator(const QString& serial_port,
-	try {
-		/* TODO: Threading code
-		std::thread x_thread(Actuator::moveActuator, m_serial_port, m_x_device, dir.x, time);
-		std::thread y_thread(Actuator::moveActuator, m_serial_port, m_y_device, dir.y, time);
-		
-		x_thread.join();
-		y_thread.join();
-		*/
-		// Temp code before multithreading
-		moveActuator(m_x_device, dir.x_comp, time);
-		moveActuator(m_y_device, dir.y_comp, time);
-	}
-	catch (std::exception& e) {
-		Logger::log(e.what(), Logger::ERROR);
-		success = false;
-	}
-
-	if (success) {
-		Logger::log("Moved { " + std::to_string(dir.x_comp) + ", " + std::to_string(dir.y_comp) + " } in " + std::to_string(time) + " milliseconds.", Logger::INFO);
-	}
-	else {
-		Logger::log("The movement { " + std::to_string(dir.x_comp) + ", " + std::to_string(dir.y_comp) + " } could not be completed.", Logger::ERROR);
-	}
+void Actuator::move(Vector2i dir, int timer) {
+    bool success = true;
+    try {
+        std::future<void> x_thread(std::async(&Actuator::moveActuator, this, m_x_device, dir.x_comp, timer));
+        std::future<void> y_thread(std::async(&Actuator::moveActuator, this, m_y_device, dir.y_comp, timer));
+
+        x_thread.get();
+        y_thread.get();
+    }
+    catch (std::exception& e) {
+        Logger::log(e.what(), Logger::ERROR);
+        success = false;
+    }
+
+    if (success) {
+        Logger::log("Moved { " + std::to_string(dir.x_comp) + ", " + std::to_string(dir.y_comp) + " } in " + std::to_string(timer) + " milliseconds.", Logger::INFO);
+    }
+    else {
+        Logger::log("The movement { " + std::to_string(dir.x_comp) + ", " + std::to_string(dir.y_comp) + " } could not be completed.", Logger::ERROR);
+    }
 }
 
-void Actuator::moveActuator(const unsigned char device, const int value, const int time) {
-	try {
-		std::chrono::milliseconds sleep_step = std::chrono::milliseconds(0);
-		if (value != 0) {
-			sleep_step = std::chrono::milliseconds(time / value); // still need to test negative steps
-		}
-		char* instr = new char(CMD_SIZE + DATA_SIZE);
-
-		// Setup device and command numbers
-		instr[0] = device;
-		instr[1] = ZaberCmd::REL_MOVE;
-
-		char* data = convertDataToBytes(STEP_FACTOR);
-
-		for (int i = 0; i < DATA_SIZE; i++) {
-			instr[i + 2] = data[i];
-		}
-
-		// TODO: This is yet to be tested, sorry I don't have Zaber actuators at home :(
-		for (int i = value; i > 0; i--) {
-			if (m_serial_port->isOpen()) {
-				m_serial_port->write(instr, CMD_SIZE + DATA_SIZE);
-			}
-			else {
-				Logger::log("ERROR: Failed to write to serial port " + (m_serial_port->portName()).toStdString() + " because it's not open.", Logger::ERROR);
-				throw "Action could not be completed"; // TODO: Might want to figure out a better way than throwing exceptions, revisit after adding concurrency
-			}
-
-			std::this_thread::sleep_for(sleep_step);
-		}
-	}
-	catch (...) {
-		throw std::current_exception();
-	}
+void Actuator::moveActuator(const unsigned char device, const int value, const int t) {

```

Comment:
Consider renaming `t` to `timer`? Or some other variable that would say more about what the parameter is used for. We should probably start using docstrings soon 😅